packages/squiggle-lang/src/rescript/Reducer/Reducer_Expression/Reducer_Expression.res

```diff
@@ -0,0 +1,90 @@
+module BuiltIn = Reducer_Dispatch_BuiltIn
+module ExpressionValue = ReducerInterface.ExpressionValue
+module Extra = Reducer_Extra
+module MathJs = Reducer_MathJs
+module Result = Belt.Result
+module T = Reducer_Expression_T
+open Reducer_ErrorValue
+
+type expression = T.expression
+type expressionValue = ExpressionValue.expressionValue
+
+/*
+  Shows the Lisp Code as text lisp code
+*/
+let rec show = expression =>
+  switch expression {
+  | T.EList(aList) =>
+    `(${Belt.List.map(aList, aValue => show(aValue))
```

Comment:
Does this all really need to be in a `(${})` string notation? I'd assume if you really needed it to be a string, you could call a function at the end.

Reply:
I am simply putting the whole string in to parentheses. There is nothing about a function call there.
We are in fact converting LISP AST to string.
list{1,2,3} in LISP would be (1 2 3).

And while debugging the reducer expressions, I am really reading it as LISP source code.  To keep it correct all the LISP semantics has to hold. They are not just any expressions or any tree of nodes.

The process is parse tree is in fact converted to LISP AST. And it is not LISP like at all. It is LISP. However the way we execute it is LISP-like not exactly LISP.

Reply:
Example,
 [0, 1, 2][1] becomes ($atIndex (0 1 2) 1). ($atIndex (0 1 2) 1) is what I need to see to verify/test that conversion is correct.

Reply:
Got it, that makes sense. I was thinking the parantheses were part of the function call, my bad!

Reply:
RESOLVED